### PR TITLE
Channel pager implementation

### PR DESCRIPTION
**Pull request by neha-Gupta1**
Pager implementation for channel and message pager.

#### Does this PR need a docs update or release note?

- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual

### CODE REVIEW

File: src/pkg/services/m365/api/channels_pager.go [hunk 1]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
 // ---------------------------------------------------------------------------
 
 type ChannelDeltaEnumerator interface {
-	DeltaGetPager
+	PageLinker
 	ValuesInPageLinker[models.Channelable]
 	SetNextLinker
 }
 
 // TODO: implement
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:91]:
```suggestion
```

File: src/pkg/services/m365/api/channels_pager.go [hunk 2]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
-// var _ ChannelDeltaEnumerator = &channelsPageCtrl{}
+var _ ChannelDeltaEnumerator = &channelPageCtrl{}
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:92]: Channels does not use a delta query query, and thus should not be named `delta` in any location.
```suggestion
var _ ChannelEnumerator = &channelPageCtrl{}
```

File: src/pkg/services/m365/api/channels_pager.go [hunk 3]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
+}
+
+func (c Channels) NewChannelPager(
+	teamID,
+	channelID string,
+	fields []string,
+) *channelPageCtrl {
+	requestConfig := &teams.ItemChannelsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsRequestBuilderGetQueryParameters{
+			Select: fields,
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:107]: Same questions as for messages: do we have any reason to select fields, and do we need to prefer the page size?

File: src/pkg/services/m365/api/channels_pager.go [hunk 4]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
+
+// ---------------------------------------------------------------------------
+// non delta item pager
+// ---------------------------------------------------------------------------
+
+type MessageItemIDType struct {
+	ItemID string
+}
+
+type channelItemPageCtrl struct {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:160]:
```suggestion
type channelMessagePageCtrl struct {
```

File: src/pkg/services/m365/api/channels_pager.go [hunk 5]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
+}
+
+func (p *channelPageCtrl) GetOdataNextLink() *string {
+	// No next link preent in the API result
+	emptyString := ""
+	return &emptyString
+}
+
+// ---------------------------------------------------------------------------
+// non delta item pager
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:153]: This should get moved up to co-locate with the other message pager.  The current page layout goes `messages > channels > messages`, which doesn't flow very well.
```suggestion
// non delta channel message pager
```

File: src/pkg/services/m365/api/channels_pager.go [hunk 6]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
+	ItemID string
+}
+
+type channelItemPageCtrl struct {
+	gs      graph.Servicer
+	builder *teams.ItemChannelsItemMessagesRequestBuilder
+	options *teams.ItemChannelsItemMessagesRequestBuilderGetRequestConfiguration
+}
+
+func (c Channels) GetItemIDsInContainer(
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:166]: This pattern is an artifact of the drive pager from which you took the boilerplate.  The drive backup system has a basis for enumerating only the IDs.  But I'm not sure there's any point to this when enumerating channel messages.  A simpler implementation is to have the non-delta enumeration return the complete chanMessageable, much like the delta enumeration does.

File: src/pkg/services/m365/api/channels_pager.go [hunk 7]
```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
...
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages()
+
+	return &channelItemPageCtrl{c.Stable, builder, options}
+}
+
+//lint:ignore U1000 False Positive
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/channels_pager.go:212]: is this necessary, or can it be removed?

**Review by ryanfkeepers — CHANGES_REQUESTED** [review]
All functions in this change need test coverage.

File: src/pkg/services/m365/api/channels_pager.go (hunk 1 above)
Comment (neha-Gupta1) [reply src/pkg/services/m365/api/channels_pager.go]: Implemented

File: src/pkg/services/m365/api/channels_pager.go (hunk 7 above)
Comment (neha-Gupta1) [reply src/pkg/services/m365/api/channels_pager.go:194]: Yes. Its required. Linter fails because of it.

**Comment by ryanfkeepers** [comment]
closing: changes were ported to other PRs

**Pull request closed without merging**